### PR TITLE
added lesson 4 file, completed homework 23_08, mod .gitignore

### PR DESCRIPTION
**Pull request by sadykovpavlo**

### CODE REVIEW

File: lesson_04/homework_04.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+""" Замініть .... на пробіл
+"""
+
+replaced_points_text = replaced_paragraph_text.replace("....", " ")
+
+# task 03 ==
+""" Зробіть так, щоб у тексті було не більше одного пробілу між словами.
+"""
+striped_text = replaced_points_text.replace("  ", "")
+
```
Comment (edanilovets) [inline lesson_04/homework_04.py]: you can use the following to get rid of extra spaces between words:
```
cleaned_sentence = " ".join(sentence.split())
```

File: lesson_04/homework_04.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+# task 05
+""" Виведіть, скільки слів у тексті починається з Великої літери?
+"""
+
+# use for to count capital letters
+counted_capital_letter = 0
+for char in text_final_version:
+    if char.isupper():
+        counted_capital_letter += 1
+
```
Comment (edanilovets) [inline lesson_04/homework_04.py]: Works for the task, but this version will also count the big letters in the middle of the word if there some, like "iPhone, eBay". The task was to count number of words that start with the upper letter.

File: lesson_04/homework_04.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+print(f"The text consist {counted_capital_letter} capital letter")
+
+# task 06
+
+""" Виведіть позицію, на якій слово Tom зустрічається вдруге
+"""
+
+print(f"Word 'Tom' position: {adventures_of_tom_sawyer.find('Tom', 2)}")
```
Comment (edanilovets) [inline lesson_04/homework_04.py]: What if don't know that text starts with Tom?
try to use find() twice

File: lesson_04/homework_04.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+# task 07
+""" Розділіть змінну adwentures_of_tom_sawer по кінцю речення.
+Збережіть результат у змінній adwentures_of_tom_sawer_sentences
+"""
+# here no print, result is list and I use variable adventures_of_tom_sawyer_sentences in the next task
+adventures_of_tom_sawyer_sentences = text_final_version.split('.')
+# KOSTYL after separate text by point at the end list has empty element "" I decided to delete it from list by del()
+del(adventures_of_tom_sawyer_sentences[-1])
```
Comment (edanilovets) [inline lesson_04/homework_04.py]: you can split like following:
```
sentence.split(". ")
```
the end of the sentece is usually dot with a space

File: lesson_04/homework_04.py (hunk 2 above)
Comment (sadykovpavlo) [reply lesson_04/homework_04.py]: fixed, added list with words and boolean method changed to istitle()

File: lesson_04/homework_04.py (hunk 3 above)
Comment (sadykovpavlo) [reply lesson_04/homework_04.py]: fixed

**Review by edanilovets — APPROVED** [review]

**Pull request merged**